torch/_jit_internal.py

```diff
@@ -663,6 +663,37 @@ def is_final(ann):
         return isinstance(ann, FinalInstance)
 
 
+try:
+    from typing import TypeVar, Generic
+
+    T = TypeVar('T')
+
+    class RRef(Generic[T]):
+        __slots__ = ['__args__']
+
+        def __init__(self, types):
+            self.__args__ = types
+
+    def is_rref(ann):
+        return getattr(ann, "__origin__", None) is RRef
+
+except ImportError:
+    class RRefInstance(object):
```

Comment:
`typing` should always be installed alongside PyTorch so I think you can get rid of the whole try catch and `RRefInstance`

Reply:
I think it's still not installed in some CI, which is blocked me landing https://github.com/pytorch/pytorch/pull/27637, I will figure out how to fix that CI and submit a separate PR to remove the whole try catch along with other cases.